### ISSUE TITLE
`glob!` on Windows doesn't seem to include `input_file`

### ISSUE THREAD

**Issue by max-sixty** [mitsuhiko/insta#336]
### What happened?

Thanks for the recent changes in 1.26! I've updated PRQL with the latest changes, they're great.

The last thing I'm having trouble with is forcing updates — Windows seems to produce different metadata, which causes its tests to fail. Specifically, Windows doesn't seem to show `input_file` in `glob!` tests. Even if we skipped those tests, anyone contributing on Windows would be submitting PRs with huge diffs.

I looked around the insta code, but couldn't find why that might happen (looks like there isn't much platform-specific code in `insta`, so it's a puzzle, and unfortunately I don't have a Windows machine, so it's difficult to debug). 

Here's an example: https://github.com/PRQL/prql/actions/runs/3861652619/jobs/6582754606#step:3:2726. (Note I made the tests pass there to see the diff, but otherwise it fails; e.g. [here](https://github.com/PRQL/prql/actions/runs/3861420435/jobs/6582387689))

Any thoughts on why that might be happening? If not I can look more.

### Insta Version

1.26

### rustc Version

rustc 1.65.0 (897e37553 2022-11-02)

**Comment by mitsuhiko** [comment]
I’m also confused why it writes .new files. Seems like it does not manage to detect the CI environment.

**Comment by mitsuhiko** [comment]
Unfortunately I am not sure what is happening here.

**Comment by max-sixty** [comment]
I just took another look. 

One of the issues is the line-endings:

```diff
-expression: "table newest_employees = (\n  from employees\n  sort tenure\n  take 50\n)\n\ntable average_salaries = (\n  from salaries\n  group country (\n    aggregate average_country_salary = (average salary)\n  )\n)\n\nfrom newest_employees\njoin average_salaries [==country]\nselect [name, salary, average_country_salary]\n"
-input_file: book/tests/prql/examples/cte-0.prql
+expression: "table newest_employees = (\r\n  from employees\r\n  sort tenure\r\n  take 50\r\n)\r\n\r\ntable average_salaries = (\r\n  from salaries\r\n  group country (\r\n    aggregate average_country_salary = (average salary)\r\n  )\r\n)\r\n\r\nfrom newest_employees\r\njoin average_salaries [==country]\r\nselect [name, salary, average_country_salary]\r\n"
```

But then the really confusing part is leaving out the `input_file` line:

```diff
diff --git a/book/tests/snapshots/snapshot__run_display_reference_prql@transforms__window-1.prql.snap b/book/tests/snapshots/snapshot__run_display_reference_prql@transforms__window-1.prql.snap
index afffeb8..c0eb1a5 100644
--- a/book/tests/snapshots/snapshot__run_display_reference_prql@transforms__window-1.prql.snap
+++ b/book/tests/snapshots/snapshot__run_display_reference_prql@transforms__window-1.prql.snap
@@ -1,7 +1,6 @@
 ---
 source: book/tests/snapshot.rs
 expression: prql_to_pl(&prql).and_then(pl_to_prql).unwrap()
-input_file: book/tests/prql/transforms/window-1.prql
 ---
 from orders
 sort day
```

Possibly someone with a windows machine would be well placed to debug...

**Comment by mitsuhiko** [comment]
The windows newlines are addressed by 48f044e0d60a2e9286567eab924b7d62a8ae5578
